flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/TableEnvironment.java

```diff
@@ -718,6 +718,36 @@ void createTemporarySystemFunction(
      */
     void createTemporaryTable(String path, TableDescriptor descriptor);
 
+    /**
+     * Registers the given {@link TableDescriptor} as a temporary catalog table.
+     *
+     * <p>The {@link TableDescriptor descriptor} is converted into a {@link CatalogTable} and stored
+     * in the catalog.
+     *
+     * <p>Temporary objects can shadow permanent ones. If a permanent object in a given path exists,
+     * it will be inaccessible in the current session. To make the permanent object available again
```

Comment:
for sentences `If a permanent object in a given path exists, it will be inaccessible in the current session. To make the permanent object available again one can drop the corresponding temporary object.` 
- can we use `table` or `TableDescriptor ` instead of `object`?
- is this true if ignoreIfExists is set.

Reply:
- Same as https://github.com/apache/flink/pull/25704#discussion_r1862771988
- Good question. I'm also not sure and would need to double-check